src/order.rs

```diff
@@ -1,22 +1,17 @@
 //! Contains logic and type definitions for orders
 use std::fmt;
-use std::mem;
 
-use byte_slice_cast::AsByteSlice;
-use byteorder::{LittleEndian, WriteBytesExt};
 use chrono::serde::ts_seconds;
 use chrono::{DateTime, Utc};
 use derive_more::Display;
 use serde::{Deserialize, Serialize};
-use sha3::{Digest, Keccak256};
 use thiserror::Error;
-use web3::signing::recover;
-use web3::types::{Address, Recovery, U256};
+use web3::types::{Address, U256};
 
 use crate::util::{from_hex_de, from_hex_se};
 
 /// Magic string representing the function signature
-pub const FUNCTION_SIGNATURE: &str = "LimitOrder(uint256 amount,uint256 price,bool side,address user,uint256 expiration,address targetTracer,uint256 nonce)";
+pub const FUNCTION_SIGNATURE: &str = "LimitOrder(uint256 amount,uint256 price,bool side,address user,uint256 expiration,address target_tracer,uint256 nonce)";
```

Comment:
I think this should still stay as targetTracer, if this is the contract representation @sporejack